File: app/Models/Enseignant.php
<?php

namespace App\Models;

use Illuminate\Notifications\Notifiable;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Database\Eloquent\Model;

class Enseignant extends Authenticatable {
    use Notifiable;

    protected $table = 'enseignants';

    protected $fillable = [
        'nomEns',
        'emailEns',
        'mdpEns',
    ];

    protected $hidden = [

    ];
}

File: lib/models/enseignant.dart
class Enseignant {
  final int? id;
  final String nomEns;
  final String emailEns;
  final String mdpEns;

  Enseignant(
      {this.id,
      required this.nomEns,
      required this.emailEns,
      required this.mdpEns});

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'nomEns': nomEns,
      'emailEns': emailEns,
      'mdpEns': mdpEns
    };
  }

  factory Enseignant.fropMap(Map<String, dynamic> map) {
    return Enseignant(
        id: map['id'],
        nomEns: map['nomEns'],
        emailEns: map['emailEns'],
        mdpEns: map['mdpEns']);
  }

  factory Enseignant.fropJson(Map<String, dynamic> json) {
    return Enseignant(
        id: json['id'],
        nomEns: json['nomEns'],
        emailEns: json['emailEns'],
        mdpEns: json['mdpEns']);
  }
}
